ipykernel/ipkernel.py

```diff
@@ -54,8 +54,7 @@ def __init__(self, **kwargs):
         # TMP - hack while developing
         self.shell._reply_content = None
 
-        self.comm_manager = CommManager(shell=self.shell, parent=self,
-                                        kernel=self)
+        self.comm_manager = CommManager(kernel=self)
```

Comment:
parent passes config. Both parent and kernel should be passed, because it is not required for `parent` to be the kernel, only a Configurable.